### PR TITLE
ENH: Remaining DBR types

### PR DESCRIPTION
**Pull request by klauer**
* DBR_GR_{*}
* DBR_STSACK_STRING
* DBR_CLASS_NAME

The framework was there, but not support in the server/payload-handling functions. This should cover the rest.

... questionable design decisions included as usual (and subject to negotiation)

... and failing travis due to `ca_test` not being there - will try again some other time.

### CODE REVIEW

**Comment by tacaswell** [comment]
I wonder if ccache + travis caching + just build epics-base would save any time?

Of course that means understanding epics build well enough to run it on travis.

**Comment by klauer** [comment]
epics-base is rather easy to build. Once you get into the realm of trying to build the motor module and areaDetector and such on top of it is where you get into trouble.

**Comment by tacaswell** [comment]
it would also let us get out of the sudo required pool which gets faster starts and (I think) better machines.

**Comment by hhslepicka** [comment]
I have being doing this for quite a while here at work! Let me know if I
can be of any help! :)

**Comment by tacaswell** [comment]
@hhslepicka can you make travis build epics-base?  We probably want to (eventually) do a matrix of libca versions.

**Comment by hhslepicka** [comment]
@tacaswell I can do that! 👍

**Comment by danielballan** [comment]
It looks like catvs runs Travis against a matrix of different branches of epics-base. See https://github.com/danielballan/catvs/blob/master/build-deps.sh and the .travis.yml.

Also, that repo has no setup.py... maybe we can send a PR in return for the free help.

**Comment by klauer** [comment]
Rebased this old one, but it will still need some work.  I don't think it'd be wise for me to do it with how my head feels, so I'll leave this for another day.

**Comment by danielballan** [comment]
Force-pushed another rebase.

File: caproto/_dbr.py [hunk 1]
```diff
@@ -1072,7 +1098,9 @@ def promote_type(ftype, *, use_status=False, use_time=False, use_ctrl=False):
         ftype += ChType.TIME_STRING
     elif use_status:
         ftype += ChType.STS_STRING
+    elif use_gr:
```
Comment (tacaswell) [inline caproto/_dbr.py]: This code block should get some comment about the assumption that the types are in blocks, all in the same order, and than the string type is the first one in each block.

Comment (danielballan) [inline caproto/_dbr.py]: Good point -- it took me a second to grok that too. Addressed.

**Comment by danielballan** [comment]
Well, we now support all the DBR types. :tada:

**Pull request merged**